lumicks/pylake/tests/data/mock_confocal.py

```diff
@@ -186,31 +211,344 @@ def make_slice(data):
+    )
+
+    times_ref = TimeStampsRef(
+        data=timestamps,
+        timestamp_ranges=timestamp_ranges,
+        dt=dt,
+        pixel_time_seconds=dt * samples_per_pixel * 1e-9,
+        line_time_seconds=dt * (pixels_per_line * samples_per_pixel + line_padding * 2) * 1e-9,
+    )
+
+    infowave_ref = InfoWaveRef(
+        data=confocal_file.infowave,
         samples_per_pixel=samples_per_pixel,
         line_padding=line_padding,
     )
 
-    return Scan(name, confocal_file, start, stop, metadata)
+    metadata_ref = MetaDataRef(
+        number_of_frames=number_of_frames,
+        lines_per_frame=lines_per_frame,
+        pixels_per_line=pixels_per_line,
+        fast_axis={0: "X", 1: "Y", 2: "Z"}[axes[0]],
+        num_pixels=[axes.num_pixels for axes in metadata.ordered_axes],
+        pixelsize_um=[axes.pixel_size_um for axes in metadata.ordered_axes],
+        center_point_um=metadata.center_point_um,
+    )
+
+    ref = ConfocalRef(
+        confocal_class=confocal_class,
+        image=image if multi_color else np.repeat(image, 3).reshape(*image.shape, 3),
+        timestamps=times_ref,
+        infowave=infowave_ref,
+        metadata=metadata_ref,
+        start=start,
+        stop=start + len(confocal_file.infowave) * dt,
+    )
+
+    return ref
+
+
+def generate_timestamps(
+    number_of_frames,
+    lines_per_frame,
+    pixels_per_line,
+    start,
+    dt,
+    samples_per_pixel,
+    line_padding,
+    scan=False,
+    x_axis_fast=True,
+):
+    """Calculate reference timestamps of a kymo or scan created with `MockConfocal.from_image()`
+
+    Parameters
+    ----------
+    number_of_frames : int
+        The number of frames to create timestamps for.
+    lines_per_frame : int
+        The number of lines per frame to create timestamps for.
+    pixels_per_line : int
+        The number of pixels per line to create timestamps for.
+    start : int
+        The start time.
+    dt : int
+        The increment for each timestamp.
+    samples_per_pixel : int
+        The number of timestamps to be used to calculate the mean timestamp for one pixel.
+    line_padding : int
+        The number of (finally unused) timestamps added to each and of one line of timestamps.
+    scan : bool
+        Create frame timestamp ranges instead of line timestamp ranges. Defaults to False. If
+        `number_of_frames` > 1 this will default to True.
+    x_axis_fast : bool
+        The x axis of the returned timestamps array is the fast indexed one (default). If False, the
+        y axis is the fast indexed and the x axis the slow indexed one.
+
+    Returns
+    -------
+    tuple(np.ndarray, np.ndarray)
+        The the pixel timestamps and the line (kymo) or frame (scan) timestamp ranges.
+    """
+    if start < 0:
+        raise ValueError("`start` needs to be a non negative int")
+    if dt <= 0:
+        raise ValueError("`dt` needs to be a positive int")
+    start = int(start)
+    dt = int(dt)
```

Comment:
This doesn't protect against roundoff that happens on creation though? If you specify the input argument as a float, then the damage has already been done at this point. It's probably better to just assert on whether they're integer, no?